ocaml/idl/datamodel_pool.ml

```diff
@@ -1024,6 +1024,42 @@ let set_uefi_certificates =
       ]
     ~allowed_roles:_R_POOL_ADMIN ()
 
+let update_sync_frequency =
+  Enum
+    ( "update_sync_frequency"
+    , [
+        ("daily", "Indicates the pool update synchronization is scheduled daily")
+      ; ( "weekly"
+        , "Indicates the pool update synchronization is scheduled weekly"
+        )
+      ; ( "monthly"
+        , "Indicates the pool update synchronization is scheduled monthly"
+        )
+      ]
+    )
+
+let configure_update_sync =
+  call ~name:"configure_update_sync"
+    ~doc:"config scheduled job to sync update from remote CDN" ~lifecycle:[]
+    ~params:
+      [
+        (Ref _pool, "self", "The pool")
+      ; ( update_sync_frequency
+        , "update_sync_frequency"
+        , "of type Enum ('daily', 'weekly', 'monthly') to record the frequency \
+           of the scheduled update synchronization"
+        )
+      ; ( Int
```

Comment:
Making day and hour separate parameters which in addition are restricted in range forces a lot of checking. There is also the problem that the legal day range depends on the frequency: on a weekly basis the day is either 0 to 6 or 1 to 7, but larger on a monthly basis. Would it have been simpler to collapse this all into one string?

* `daily:13:00` (every day at 13:00)
* `weekly:3-13:30` (every week on, Wednesday on 13:30)
* `monthly:15-14:45` (every 15th, on 14:45)
Parsing a string like this and finding illegal strings would be one compact function and a single error.

Reply:
@lindig Thanks for the suggestion.
I think this API design was chosen in the design: [Design of Toolstack for CP-41451](https://info.citrite.net/display/xenserver/Design+of+Toolstack+for+CP-41451) is because we can do parameter checking easily.
And with your suggested string style parameter, is there a simple way to tell both the API user and xe CLI user the expected format of this string?

Reply:
I thought a bit more about it. I think it's fine as is. Using a single string is an alternative design (a domain-specific format) ; the important part is that all parameters are set in one call and not individually because the legal value ranges depend on each other.